### PR TITLE
Adds privacy contracts

### PR DESCRIPTION
**Pull request by josh-richardson**
Signed-off-by: Joshua Richardson <joshua@richardson.tech>

### CODE REVIEW

**Comment by josh-richardson** [comment]
This should now be ready to go.

File: acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/transaction/DeploySmartContractTransaction.java [hunk 1]
```diff
@@ -33,23 +36,55 @@
...
 
   @Override
+  @SuppressWarnings("rawtypes")
   public T execute(final NodeRequests node) {
     try {
-      final Method method =
-          clazz.getMethod(
-              "deploy", Web3j.class, Credentials.class, BigInteger.class, BigInteger.class);
+      if (args != null && args.length != 0) {
+        ArrayList<Object> parameterObjects = new ArrayList<>();
```
Comment (iikirilov) [inline acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/transaction/DeploySmartContractTransaction.java]: nit: final

File: acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/transaction/DeploySmartContractTransaction.java [hunk 2]
```diff
@@ -33,23 +36,55 @@
...
+        parameterObjects.addAll(
+            Arrays.asList(node.eth(), BENEFACTOR_ONE, DEFAULT_GAS_PRICE, DEFAULT_GAS_LIMIT));
+        parameterObjects.addAll(Arrays.asList(args));
 
-      final Object invoked =
-          method.invoke(
-              METHOD_IS_STATIC, node.eth(), BENEFACTOR_ONE, DEFAULT_GAS_PRICE, DEFAULT_GAS_LIMIT);
+        ArrayList<Class> parameterTypes =
```
Comment (iikirilov) [inline acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/transaction/DeploySmartContractTransaction.java]: nit: final

File: acceptance-tests/tests/build.gradle [hunk 3]
```diff
@@ -20,12 +20,14 @@ dependencies {
...
   testImplementation project(':ethereum:core')
   testImplementation project(':ethereum:rlp')
+  testImplementation project(':privacy-contracts')
   testImplementation project(path: ':ethereum:core', configuration: 'testSupportArtifacts')
   testImplementation project(':ethereum:permissioning')
   testImplementation project(':plugin-api')
   testImplementation project(':testutil')
   testImplementation project(':util')
 
+
```
Comment (iikirilov) [inline acceptance-tests/tests/build.gradle]: nit: new line

File: acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/web3j/privacy/contracts/PrivacyGroupTest.java [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        .isEqualTo(privacyGroup.getParticipants(firstParticipant.raw()).encodeFunctionCall());
+    // 0xf744b089
+    assertThat(
+            "0xf744b089035695b4cc4b0941e60551d7a19cf30603db5bfc23e5ac43a56f57f25f75486a000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000012a8d9b56a0fe9cd94d60be4413bcb721d3a7be27ed8e28b3a6346df874ee141b")
+        .isEqualTo(
+            privacyGroup
+                .addParticipants(
+                    firstParticipant.raw(), Collections.singletonList(secondParticipant.raw()))
+                .encodeFunctionCall());
+    assertThat("0xf83d08ba").isEqualTo(privacyGroup.lock().encodeFunctionCall());
```
Comment (iikirilov) [inline acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/web3j/privacy/contracts/PrivacyGroupTest.java]: Please can we make static fields for all of the hex strings in this class.

Maybe you can reuse `OnChainGroupManagement`

File: acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/web3j/privacy/contracts/PrivacyProxyTest.java [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    assertThat(privacyProxy.isValid()).isEqualTo(true);
+    contractVerifier
+        .validTransactionReceipt(privacyProxy.getContractAddress())
+        .verify(privacyProxy);
+    // 0x0b0235be
+    assertThat("0x0b0235bef772b2ee55f016431cefe724a05814324bb96e9afdb73e338665a693d4653678")
+        .isEqualTo(privacyProxy.getParticipants(firstParticipant.raw()).encodeFunctionCall());
+    // 0xf744b089
+    assertThat(
+            "0xf744b089f772b2ee55f016431cefe724a05814324bb96e9afdb73e338665a693d465367800000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000000000")
```
Comment (iikirilov) [inline acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/web3j/privacy/contracts/PrivacyProxyTest.java]: hex to static fields please

**Review by iikirilov — CHANGES_REQUESTED** [review]

**Comment by josh-richardson** [comment]
Comments addressed.

**Pull request merged**